opm/material/densead/DynamicEvaluation.hpp

```diff
@@ -0,0 +1,479 @@
+    {
+        // values are divided, derivatives follow the rule for division, i.e., (u/v)' = (v'u -
+        // u'v)/v^2.
+        ValueType& u = data_[valuepos_()];
+        const ValueType& v = other.value();
+        for (int idx = dstart_(); idx < dend_(); ++idx) {
+            const ValueType& uPrime = data_[idx];
+            const ValueType& vPrime = other.data_[idx];
+
+            data_[idx] = (v*uPrime - u*vPrime)/(v*v);
+        }
+        u /= v;
+
+        return *this;
+    }
+
+    // divide value and derivatives by value of other
+    template <class RhsValueType>
+    Evaluation& operator/=(const RhsValueType& other)
+    {
+        const ValueType tmp = 1.0/other;
+
+        for (int i = 0; i < length_(); ++i)
+            data_[i] *= tmp;
+
+        return *this;
+    }
+
+    // add two evaluation objects
+    Evaluation operator+(const Evaluation& other) const
+    {
+        Evaluation result(*this);
+
+        result += other;
+
+        return result;
+    }
+
+    // add constant to this object
+    template <class RhsValueType>
+    Evaluation operator+(const RhsValueType& other) const
+    {
+        Evaluation result(*this);
+
+        result += other;
+
+        return result;
+    }
+
+    // subtract two evaluation objects
+    Evaluation operator-(const Evaluation& other) const
+    {
+        Evaluation result(*this);
+
+        result -= other;
+
+        return result;
+    }
+
+    // subtract constant from evaluation object
+    template <class RhsValueType>
+    Evaluation operator-(const RhsValueType& other) const
+    {
+        Evaluation result(*this);
+
+        result -= other;
+
+        return result;
+    }
+
+    // negation (unary minus) operator
+    Evaluation operator-() const
+    {
+        Evaluation result;
+
+        // set value and derivatives to negative
+        for (int i = 0; i < length_(); ++i)
+            result.data_[i] = - data_[i];
+
+        return result;
+    }
+
+    Evaluation operator*(const Evaluation& other) const
+    {
+        Evaluation result(*this);
+
+        result *= other;
+
+        return result;
+    }
+
+    template <class RhsValueType>
+    Evaluation operator*(const RhsValueType& other) const
+    {
+        Evaluation result(*this);
+
+        result *= other;
+
+        return result;
+    }
+
+    Evaluation operator/(const Evaluation& other) const
+    {
+        Evaluation result(*this);
+
+        result /= other;
+
+        return result;
+    }
+
+    template <class RhsValueType>
+    Evaluation operator/(const RhsValueType& other) const
+    {
+        Evaluation result(*this);
+
+        result /= other;
+
+        return result;
+    }
+
+    template <class RhsValueType>
+    Evaluation& operator=(const RhsValueType& other)
+    {
+        setValue( other );
+        clearDerivatives();
+
+        return *this;
+    }
+
+    // copy assignment from evaluation
+    Evaluation& operator=(const Evaluation& other) = default;
```

Comment:
All the operators between the `Evaluation`s we should make sure they have the same number of Primary Variables.

This is something new compared with the fixed-size `Evaluation`. 

Maybe `assert` will be proper in case we care about the performance of the release running.

Reply:
good point, PR updated. Note that in practice the difference is rather small as non-matching dynamic evaluations will most likely lead to segmentation faults.